### ISSUE TITLE
Create ingestion engine appliances(s)

### ISSUE THREAD

**Issue by chriszs** [tvkitchen/appliances#32]
#### Description
Now that we've decided ingestion engines are appliances, we need to port them over from the `tv-kitchen` repo. Presumably we want one appliance per engine. Where does the abstract ingestion appliance go? Let's find out.

#### Related Issues
tvkitchen/meta/issues/14

**Comment by slifty** [comment]
I would recommend putting the abstract ingestion appliance in the appliance-core package within the `appliances` repo as a second export.

**Comment by chriszs** [comment]
Says @slifty, who has definitely already planned this whole PR in his head.

**Comment by chriszs** [comment]
And in [this comment](https://github.com/tvkitchen/meta/issues/14#issuecomment-658337276) I only half-read before where he lays out what he just said.

**Comment by slifty** [comment]
Now you got me all nervous 😂 

Here's a guiding philosophy I would propose here: `AbstractAppliance` and `AbstractIngestionAppliance` should be peers.

There are a few ways that we could make them peers:

## 1. Peers as exports from the `appliance-core` package.

We create a new AbstractIngestionAppliance class within the `appliance-core` package inside of the `appliances` repository.

In this model an import would look like this:

```
import { AbstractIngestionAppliance } from '@tvkitchen/appliance-core'
```

## 2. Peers as packages in the `appliances` repository.

We create a new `ingestion-appliance-core` (name TBD) package inside of the `appliances` repository.

In this model an import would look like this:

```
import AbstractIngestionAppliace from @tvkitchen/appliance-ingestion-core'
```

### 3. Peers as classes within `base-classes`.

Here we would *remove* the `appliance-core` package entirely, move `AbstractAppliance` to `base-classes`, and create a `AbstractIngestionAppliance` class in `base-classes` as well.

In this model appliances all either
```
import { AbstractAppliance } from '@tvkitchen/base-classes'
```

or

```
import { AbstractIngestionAppliance } from '@tvkitchen/base-classes'
````

----

I should note that AbstractAppliance has NOT been published to NPM, so we are not impacted by semver considerations yet.

I think we should think hard about whether `appliance-core` was the right place for `AbstractAppliance` as opposed to `base-classes`. 

The original reasoning was that AbstractAppliance was not needed in the countertop or consuming APIs since type checking is already covered by `IAppliance`.

I think we want to keep `base-classes` for code and classes that are truly relevant across all aspects of the TV Kitchen ecosystem (e.g. payloads, payload arrays)

This means that `appliance-core` (name can be changed to `appliance-base-classes` tbh!) is really appliance specific and would NOT be imported by anything that is not an appliance implementation.

---

TL;DR I think I'm still leaning on option 1 (with the potential desire to also rename `appliance-core` to something more meaningful.)
